mypy/checker.py

```diff
@@ -3851,6 +3868,20 @@ def visit_decorator(self, e: Decorator) -> None:
             if e.type and not isinstance(get_proper_type(e.type), (FunctionLike, AnyType)):
                 self.fail(message_registry.BAD_CONSTRUCTOR_TYPE, e)
 
+    def get_decorator_fullname_and_object_type(
```

Comment:
Is there a reason to not keep this method protected?

```suggestion
    def _get_decorator_fullname_and_object_type(
```

Reply:
I 100% agree, but mypy does not use protected methods that much. So, almost everything is "public". I prefer to follow the style here 🙂